src/components/Articles/ArticleSort.jsx

```diff
@@ -0,0 +1,24 @@
+export const ArticleSort = ({ searchParams, setSearchParams }) => {
+  const handleChange = (e) => {
+    const newParams = new URLSearchParams(searchParams);
+    newParams.set(e.target.name, e.target.value);
+    setSearchParams(newParams);
```

Comment:
I'm unclear why you're reassigning the searchParams in this handle change. I'd expect your drop down to amend the url which it is doing. Your use effect should have the topic, sort_by and order as dependancies which I believe would negate the need for this. 

However if you've followed documentation to do this I would be very interested to read it.